subprojects/antlr/src/main/java/org/gradle/api/plugins/antlr/AntlrPlugin.java

```diff
@@ -80,7 +89,7 @@ public Object call() throws Exception {
             }
         });
 
-        project.getConvention().getPlugin(JavaPluginConvention.class).getSourceSets().all(
+        project.getConvention().getPlugin(JavaPluginConvention.class).getSourceSets().configureEach(
```

Comment:
This shouldn't be lazy. It needs to be eagerly evaluated because a task needs to be created for each source set.